gcp-observability/src/main/java/io/grpc/gcp/observability/ObservabilityConfig.java

```diff
@@ -54,27 +55,49 @@ public interface ObservabilityConfig {
   /**
    * POJO for representing a filter used in configuration.
    */
+  @ThreadSafe
   class LogFilter {
-    /** Pattern indicating which service/method to log. */
-    public final String pattern;
+    /** Pattern for service/method filter. */
+    public final List<String> pattern;
 
-    /** Number of bytes of each header to log. */
+    /** Boolean to indicate all services and methods. */
+    public final Boolean matchAll;
+
+    /** Set of services. */
+    public final Set<String> services;
+
+    /* Set of fullMethodNames. */
+    public final Set<String> methods;
+
+    /** Number of bytes of header to log. */
     public final Integer headerBytes;
```

Comment:
I'd expect these to be unboxed, since they have clear defaults defined. Things like `if (logFilter.excludePattern) {` are just asking for a NPE right now.

Is matchAll in the same situation?

Reply:
I think boxed types were used, because of the return types from `JsonUtil`. But I see that using unboxed types makes sense here. I will update.